compiler/semantic.ml

```diff
@@ -57,7 +66,17 @@ let check_binop_type (left_expr : data_type) (op : Ast.math_op) (right_expr : da
 	with (Int, _, Int) -> Int
 	| (Float, _, Float) -> Float
 	| (String, Add, String) -> String
-	| (_, _, _) -> raise (Failure "cannot perform binary operations with provided arguments")
+	| (AnyType, _, Int) -> Int
+	| (AnyType, _, Float) -> Float
+	| (AnyType, Add, String) -> String
+	| (Int, _, AnyType) -> Int
+	| (Float, _, AnyType) -> Float
+	| (String, Add, AnyType) -> String
+	| (AnyType, Add, AnyType) -> AnyType
+	| (AnyType, _, AnyType) -> Float
+		(* If we're doing math on two any types, just assume it's a float. *)
```

Comment:
I know we've already merged this PR in but thoughts on making this String? Since JSON stores more strings than anything else in general it might be safest to typecast them as a String and take it from there?